helix-term/src/commands.rs

```diff
@@ -5115,11 +5115,22 @@ fn surround_replace(cx: &mut Context) {
             }
         };
 
+        // Visual feedback
+        let selection = selection.clone();
+        let mut ranges: SmallVec<[Range; 1]> = SmallVec::new();
+        // TODO: Use [`array_chunks`] once stabilized
+        for p in change_pos.chunks_exact(2) {
```

Comment:
With the latest code, do wee even need to chunk? It's just mapping all the values to point ranges now

Reply:
I wonder why I chunked in the first place... I've revisited and gone the extra mile now, making the transaction work on the selection, since we're updating it anyway to show the visual feedback. Hopefully that should be it.